draft-chen-cfrg-vdaf-pine.md

```diff
@@ -307,8 +307,33 @@ We define a function `PineValid.encode_gradient(self, measurement: list[float])
 
 ### Running the Wraparound Tests {#run-wr-check}
 
-> XXX Give an overview of how the wraparound test results are generated from
-> the XOF seed.
+Given the encoded gradient from {{encode-gradient}} and the XOF to generate the
+random vectors, the Client needs to run wraparound check for `num_wr_checks`
+times, and is required to pass at least `num_wr_successes` of them. Each
+wraparound check works as the following:
+
+The Client generates a random vector, which has the same dimension as the
+gradient `dimension`. Each entry of the random vector is a field element of `1`
+with probability `1/4`, or `0` with probability `1/2`, or `q-1` with probability
+`1/4`, over the field modulus `q`. The Client samples each entry by sampling
+from the XOF two bits each time:
+* If the bits are `00`, set the entry to be `q-1`.
+* If the bits are `01` or `10`, set the entry to be `0`.
+* If the bits are `11`, set the entry to be `1`.
+
+Then the Client computes a dot product, i.e., the sum of elementwise
+multiplications of the encoded gradient and the random vector, modulo `q`. If
+the dot product is in the range of
+`[-wr_check_bound, wr_check_bound + Field(1)]`, then the Client passes that
+wraparound check, and fails otherwise. Note the Client doesn't send this dot
```

Comment:
nit (grammar): As a general rule, technical writing doesn't have contractions in it. Some authors make exceptions of course. In an Internet-Draft, I'd probably not use contractions.
```suggestion
wraparound check, and fails otherwise. Note the Client does not send this dot
```